coucharchive

```diff
@@ -255,6 +257,35 @@ class ReplicationControl(object):
         self._last_successes = multiprocessing.Manager().list()
         self._last_errors = multiprocessing.Manager().list()
 
+        self.has_run_queue_control = has_run_queue_control
+        if self.has_run_queue_control:
+            def run_queue_control():
+                t = threading.currentThread()
+                while getattr(t, "do_run", True):
```

Comment:
Suggestion for understanding (take it or leave it):
```python
                t = threading.currentThread()
                t.do_run = True
                while t.do_run:
```

Reply:
I did not understood, do you mean something like that?
```py
                t = threading.currentThread()
                t.do_run = getattr(t, "do_run", True)
                while t.do_run:

Reply:
No, I meant the code I posted, with `t.do_run = True` set once at the beginning, then loop until another thread does `t.do_run = False`. Maybe I missed something? Otherwise, I find my proposal more readable!

Reply:
Ah yes, it works great indeed, thanks!